--- mpage-2.5.8/Makefile.orig	2023-05-02 23:53:09.000000000 +0200
+++ mpage-2.5.8/Makefile	2024-05-29 18:42:14.723401933 +0200
@@ -57,11 +57,10 @@ PAGESIZE=A4
 SPOOL_TYPE=BSD_SPOOLER
 
 # PREFIX=e:/usr		# OS/2 type
-PREFIX=/usr/local
-#PREFIX=/usr
+PREFIX=/usr
 BINDIR=$(PREFIX)/bin
 LIBDIR=$(PREFIX)/share
-MANDIR=$(PREFIX)/man/man1
+MANDIR=$(PREFIX)/share/man/man1
 
 #
 # A default encoding is given in encoding.h. Setting ENCODING=1 will
@@ -114,7 +113,7 @@ O = .o
 # If you are using gcc, you probably don't need to change anything here.
 
 # Linux:
-CFLAGS = -O2 -s $(DEFS) -Wall
+CFLAGS = ${OPT_FLAGS} $(DEFS) -Wall
 
 # AIX (xlC on aix 4):
 #CFLAGS = -O2 -s $(DEFS)
