--- mpage-2.5.1/mpage.h~	1998-03-24 23:07:58.000000000 +0100
+++ mpage-2.5.1/mpage.h	2007-02-15 21:43:42.000000000 +0100
@@ -159,7 +159,6 @@
 extern int ps_width;	/* number of points in the X direction (8.5 inches) */
 extern int ps_height;	/* number of points in the Y direction (11 inches) */
 extern char * media;	/* name of output page media */
-extern struct page_desc paper[];
 
 /*
  * a sheet describes the measurements and orientatation of a page for
@@ -208,7 +207,8 @@
     int width;
     int height;
 };
-    
+
+extern struct page_desc paper[];
 
 /* array of sheets where pages are ordered for coli ??? */
 extern struct sheet coli[];
